cove_ocds/tests.py

```diff
@@ -674,6 +674,42 @@ def test_schema_after_version_change(client):
         assert "'version' is missing but required" not in validation_errors_fp.read()
 
 
+@pytest.mark.django_db
+def test_schema_after_version_change_record(client):
+    data = SuppliedData.objects.create()
+    with open(os.path.join('cove_ocds', 'fixtures', 'tenders_records_1_record_with_invalid_extensions.json')) as fp:
+        data.original_file.save('test.json', UploadedFile(fp))
+
+    resp = client.post(data.get_absolute_url(), {'version': '1.1'})
+    assert resp.status_code == 200
+
+    #with open(os.path.join(data.upload_dir(), 'extended_record_schema.json')) as extended_record_fp:
+    #    assert "mainProcurementCategory" in json.load(extended_record_fp)['definitions']['Tender']['properties']
+
+    with open(os.path.join(data.upload_dir(), 'validation_errors-2.json')) as validation_errors_fp:
+        assert "'version' is missing but required" in validation_errors_fp.read()
+
+    # test link is still there.
+    resp = client.get(data.get_absolute_url())
+    assert resp.status_code == 200
+    #assert 'extended_record_schema.json' in resp.content.decode()
+
+    #with open(os.path.join(data.upload_dir(), 'extended_record_schema.json')) as extended_record_fp:
+    #    assert "mainProcurementCategory" in json.load(extended_record_fp)['definitions']['Tender']['properties']
+
+    with open(os.path.join(data.upload_dir(), 'validation_errors-2.json')) as validation_errors_fp:
+        assert "'version' is missing but required" in validation_errors_fp.read()
+
+    resp = client.post(data.get_absolute_url(), {'version': '1.0'})
+    assert resp.status_code == 200
+
+    #with open(os.path.join(data.upload_dir(), 'extended_record_schema.json')) as extended_record_fp:
+    #    assert "mainProcurementCategory" not in json.load(extended_record_fp)['definitions']['Tender']['properties']
+
```

Comment:
Yes, to highlight that we might want to test something like this for records, but at the moment it doesn't happen - https://github.com/OpenDataServices/cove/issues/747. Perhaps I should add a note with a link to that issue.